### ISSUE TITLE
Can't install prusa leveling guide

### ISSUE THREAD

**Issue by jec0709** [scottrini/OctoPrint-PrusaLevelingGuide#32]
Can't install prusa leveling guide
What should I do?
code : 
Updating, please wait.
/home/pi/oprint/bin/python2 -m pip --disable-pip-version-check install https://github.com/scottrini/OctoPrint-PrusaLevelingGuide/archive/1.0.13.zip --no-cache-dir
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting https://github.com/scottrini/OctoPrint-PrusaLevelingGuide/archive/1.0.13.zip
Downloading https://github.com/scottrini/OctoPrint-PrusaLevelingGuide/archive/1.0.13.zip
Requirement already satisfied: OctoPrint in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint-PrusaLevelingGuide==1.0.13) (1.5.3)
Requirement already satisfied: regex in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint-PrusaLevelingGuide==1.0.13) (2019.8.19)
Requirement already satisfied: numpy in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint-PrusaLevelingGuide==1.0.13) (1.16.6)
Collecting scipy
Downloading scipy-1.2.3.tar.gz (23.3 MB)
Requirement already satisfied: OctoPrint-FileCheck>=2020.08.07 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (2020.8.7)
Requirement already satisfied: futures<4,>=3.3; python_version < "3" in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (3.3.0)
Requirement already satisfied: semantic-version<3,>=2.8.5 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (2.8.5)
Requirement already satisfied: itsdangerous<2,>=1.1.0 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (1.1.0)
Requirement already satisfied: future<1,>=0.18.2 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (0.18.2)
Requirement already satisfied: blinker<2,>=1.4 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (1.4)
Requirement already satisfied: unidecode<0.05,>=0.04.14 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (0.4.21)
Requirement already satisfied: werkzeug<2,>=1.0.1 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (1.0.1)
Requirement already satisfied: Flask-Assets<3,>=2.0 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (2.0)
Requirement already satisfied: PyYAML<6,>=5.3.1 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (5.3.1)
Requirement already satisfied: pylru<2,>=1.2 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (1.2.0)
Requirement already satisfied: pyserial<4,>=3.4 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (3.5)
Requirement already satisfied: pkginfo<2,>=1.5.0.1 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (1.5.0.1)
Requirement already satisfied: wrapt<2,>=1.12.1 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (1.12.1)
Requirement already satisfied: chainmap<2,>=1.0.3; python_version < "3" in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (1.0.3)
Requirement already satisfied: Click<8,>=7.1.2 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (7.1.2)
Requirement already satisfied: markupsafe<2.0,>=1.1 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (1.1.1)
Requirement already satisfied: markdown<3.2,>=3.1 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (3.1.1)
Requirement already satisfied: netifaces<1,>=0.10.9 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (0.10.9)
Requirement already satisfied: scandir<2,>=1.10; python_version < "3" in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (1.10.0)
Requirement already satisfied: rsa==4.0 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (4.0)
Requirement already satisfied: filetype<2,>=1.0.7 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (1.0.7)
Requirement already satisfied: tornado==5.1.1 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (5.1.1)
Requirement already satisfied: sarge==0.1.5post0 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (0.1.5.post0)
Requirement already satisfied: feedparser<6,>=5.2.1; python_version < "3" in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (5.2.1)
Requirement already satisfied: cachelib<1,>=0.1 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (0.1)
Requirement already satisfied: OctoPrint-FirmwareCheck>=2020.09.23 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (2021.2.4)
Requirement already satisfied: typing<4,>=3.7.4.1; python_version < "3" in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (3.7.4.1)
Requirement already satisfied: flask<2,>=1.1.2 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (1.1.2)
Requirement already satisfied: sentry-sdk<1,>=0.15.1 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (0.16.3)
Requirement already satisfied: watchdog==0.10.4 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (0.10.4)
Requirement already satisfied: frozendict<2,>=1.2 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (1.2)
Requirement already satisfied: netaddr<1,>=0.7.19 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (0.7.19)
Requirement already satisfied: Jinja2<3,>=2.11.2 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (2.11.2)
Requirement already satisfied: Flask-Login<0.6,>=0.5 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (0.5.0)
Requirement already satisfied: emoji<1,>=0.5.4 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (0.5.4)
Requirement already satisfied: requests<3,>=2.23.0 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (2.24.0)
Requirement already satisfied: Flask-Babel<2,>=1.0 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (1.0.0)
Requirement already satisfied: websocket-client<1,>=0.57 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (0.57.0)
Requirement already satisfied: monotonic<2,>=1.5; python_version < "3" in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (1.5)
Requirement already satisfied: enum34<1.2,>=1.1.10; python_version < "3" in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (1.1.10)
Requirement already satisfied: psutil<6,>=5.7 in /home/pi/oprint/lib/python2.7/site-packages (from OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (5.7.2)
Requirement already satisfied: webassets>=2.0 in /home/pi/oprint/lib/python2.7/site-packages (from Flask-Assets<3,>=2.0->OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (2.0)
Requirement already satisfied: setuptools>=36 in /home/pi/oprint/lib/python2.7/site-packages (from markdown<3.2,>=3.1->OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (41.5.1)
Requirement already satisfied: pyasn1>=0.1.3 in /home/pi/oprint/lib/python2.7/site-packages (from rsa==4.0->OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (0.4.7)
Requirement already satisfied: singledispatch in /home/pi/oprint/lib/python2.7/site-packages (from tornado==5.1.1->OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (3.4.0.3)
Requirement already satisfied: backports-abc>=0.4 in /home/pi/oprint/lib/python2.7/site-packages (from tornado==5.1.1->OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (0.5)
Requirement already satisfied: certifi in /home/pi/oprint/lib/python2.7/site-packages (from sentry-sdk<1,>=0.15.1->OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (2019.9.11)
Requirement already satisfied: urllib3>=1.10.0 in /home/pi/oprint/lib/python2.7/site-packages (from sentry-sdk<1,>=0.15.1->OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (1.25.6)
Requirement already satisfied: pathtools>=0.1.1 in /home/pi/oprint/lib/python2.7/site-packages (from watchdog==0.10.4->OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (0.1.2)
Requirement already satisfied: chardet<4,>=3.0.2 in /home/pi/oprint/lib/python2.7/site-packages (from requests<3,>=2.23.0->OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (3.0.4)
Requirement already satisfied: idna<3,>=2.5 in /home/pi/oprint/lib/python2.7/site-packages (from requests<3,>=2.23.0->OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (2.8)
Requirement already satisfied: Babel>=2.3 in /home/pi/oprint/lib/python2.7/site-packages (from Flask-Babel<2,>=1.0->OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (2.7.0)
Requirement already satisfied: pytz in /home/pi/oprint/lib/python2.7/site-packages (from Flask-Babel<2,>=1.0->OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (2019.3)
Requirement already satisfied: six in /home/pi/oprint/lib/python2.7/site-packages (from websocket-client<1,>=0.57->OctoPrint->OctoPrint-PrusaLevelingGuide==1.0.13) (1.12.0)
Building wheels for collected packages: OctoPrint-PrusaLevelingGuide, scipy
Building wheel for OctoPrint-PrusaLevelingGuide (setup.py): started
Building wheel for OctoPrint-PrusaLevelingGuide (setup.py): finished with status 'done'
Created wheel for OctoPrint-PrusaLevelingGuide: filename=OctoPrint_PrusaLevelingGuide-1.0.13-py2-none-any.whl size=228838 sha256=26010c8d99796e4a1cc0894075d5f41e5ef714402fb81321d1f0deb151d97725
Stored in directory: /tmp/pip-ephem-wheel-cache-UH7egg/wheels/96/c0/ab/fb064e47157a029ca2bb775e5e521fb8ee82033be78df396fc
Building wheel for scipy (setup.py): started
ERROR: Command errored out with exit status 1:
command: /home/pi/oprint/bin/python2 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-mMxjGq/scipy/setup.py'"'"'; __file__='"'"'/tmp/pip-install-mMxjGq/scipy/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-VKEBCo
cwd: /tmp/pip-install-mMxjGq/scipy/
Complete output (143 lines):
Running from scipy source directory.
lapack_opt_info:
lapack_mkl_info:
customize UnixCCompiler
libraries mkl_rt not found in ['/home/pi/oprint/lib', '/usr/local/lib', '/usr/lib', '/usr/lib/arm-linux-gnueabihf']
NOT AVAILABLE
openblas_lapack_info:
customize UnixCCompiler
customize UnixCCompiler
libraries openblas not found in ['/home/pi/oprint/lib', '/usr/local/lib', '/usr/lib', '/usr/lib/arm-linux-gnueabihf']
NOT AVAILABLE
openblas_clapack_info:
customize UnixCCompiler
customize UnixCCompiler
libraries openblas,lapack not found in ['/home/pi/oprint/lib', '/usr/local/lib', '/usr/lib', '/usr/lib/arm-linux-gnueabihf']
NOT AVAILABLE
atlas_3_10_threads_info:
Setting PTATLAS=ATLAS
customize UnixCCompiler
libraries lapack_atlas not found in /home/pi/oprint/lib
customize UnixCCompiler
libraries tatlas,tatlas not found in /home/pi/oprint/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/local/lib
customize UnixCCompiler
libraries tatlas,tatlas not found in /usr/local/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/lib
customize UnixCCompiler
libraries tatlas,tatlas not found in /usr/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/lib/arm-linux-gnueabihf
customize UnixCCompiler
libraries tatlas,tatlas not found in /usr/lib/arm-linux-gnueabihf
<class 'numpy.distutils.system_info.atlas_3_10_threads_info'>
NOT AVAILABLE
atlas_3_10_info:
customize UnixCCompiler
libraries lapack_atlas not found in /home/pi/oprint/lib
customize UnixCCompiler
libraries satlas,satlas not found in /home/pi/oprint/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/local/lib
customize UnixCCompiler
libraries satlas,satlas not found in /usr/local/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/lib
customize UnixCCompiler
libraries satlas,satlas not found in /usr/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/lib/arm-linux-gnueabihf
customize UnixCCompiler
libraries satlas,satlas not found in /usr/lib/arm-linux-gnueabihf
<class 'numpy.distutils.system_info.atlas_3_10_info'>
NOT AVAILABLE
atlas_threads_info:
Setting PTATLAS=ATLAS
customize UnixCCompiler
libraries lapack_atlas not found in /home/pi/oprint/lib
customize UnixCCompiler
libraries ptf77blas,ptcblas,atlas not found in /home/pi/oprint/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/local/lib
customize UnixCCompiler
libraries ptf77blas,ptcblas,atlas not found in /usr/local/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/lib
customize UnixCCompiler
libraries ptf77blas,ptcblas,atlas not found in /usr/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/lib/arm-linux-gnueabihf
customize UnixCCompiler
libraries ptf77blas,ptcblas,atlas not found in /usr/lib/arm-linux-gnueabihf
<class 'numpy.distutils.system_info.atlas_threads_info'>
NOT AVAILABLE
atlas_info:
customize UnixCCompiler
libraries lapack_atlas not found in /home/pi/oprint/lib
customize UnixCCompiler
libraries f77blas,cblas,atlas not found in /home/pi/oprint/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/local/lib
customize UnixCCompiler
libraries f77blas,cblas,atlas not found in /usr/local/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/lib
customize UnixCCompiler
libraries f77blas,cblas,atlas not found in /usr/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/lib/arm-linux-gnueabihf
customize UnixCCompiler
libraries f77blas,cblas,atlas not found in /usr/lib/arm-linux-gnueabihf
<class 'numpy.distutils.system_info.atlas_info'>
NOT AVAILABLE
accelerate_info:
NOT AVAILABLE
/home/pi/oprint/local/lib/python2.7/site-packages/numpy/distutils/system_info.py:639: UserWarning:
Atlas (http://math-atlas.sourceforge.net/) libraries not found.
Directories to search for the libraries can be specified in the
numpy/distutils/site.cfg file (section [atlas]) or by setting
the ATLAS environment variable.
self.calc_info()
lapack_info:
customize UnixCCompiler
libraries lapack not found in ['/home/pi/oprint/lib', '/usr/local/lib', '/usr/lib', '/usr/lib/arm-linux-gnueabihf']
NOT AVAILABLE
/home/pi/oprint/local/lib/python2.7/site-packages/numpy/distutils/system_info.py:639: UserWarning:
Lapack (http://www.netlib.org/lapack/) libraries not found.
Directories to search for the libraries can be specified in the
numpy/distutils/site.cfg file (section [lapack]) or by setting
the LAPACK environment variable.
self.calc_info()
lapack_src_info:
NOT AVAILABLE
/home/pi/oprint/local/lib/python2.7/site-packages/numpy/distutils/system_info.py:639: UserWarning:
Lapack (http://www.netlib.org/lapack/) sources not found.
Directories to search for the sources can be specified in the
numpy/distutils/site.cfg file (section [lapack_src]) or by setting
the LAPACK_SRC environment variable.
self.calc_info()
NOT AVAILABLE
Traceback (most recent call last):
File "<string>", line 1, in <module>
File "/tmp/pip-install-mMxjGq/scipy/setup.py", line 492, in <module>
setup_package()
File "/tmp/pip-install-mMxjGq/scipy/setup.py", line 488, in setup_package
setup(**metadata)
File "/home/pi/oprint/local/lib/python2.7/site-packages/numpy/distutils/core.py", line 137, in setup
config = configuration()
File "/tmp/pip-install-mMxjGq/scipy/setup.py", line 395, in configuration
raise NotFoundError(msg)
numpy.distutils.system_info.NotFoundError: No lapack/blas resources found.
----------------------------------------
ERROR: Failed building wheel for scipy
Building wheel for scipy (setup.py): finished with status 'error'
Running setup.py clean for scipy
Successfully built OctoPrint-PrusaLevelingGuide
Failed to build scipy
Installing collected packages: scipy, OctoPrint-PrusaLevelingGuide
Running setup.py install for scipy: started
ERROR: Command errored out with exit status 1:
command: /home/pi/oprint/bin/python2 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-mMxjGq/scipy/setup.py'"'"'; __file__='"'"'/tmp/pip-install-mMxjGq/scipy/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' clean --all
cwd: /tmp/pip-install-mMxjGq/scipy
Complete output (9 lines):
`setup.py clean` is not supported, use one of the following instead:
- `git clean -xdf` (cleans all files)
- `git clean -Xdf` (cleans all versioned files, doesn't touch
files that aren't checked into the git repo)
Add `--force` to your command to use it anyway if you must (unsupported).
----------------------------------------
ERROR: Failed cleaning build dir for scipy
ERROR: Command errored out with exit status 1:
command: /home/pi/oprint/bin/python2 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-mMxjGq/scipy/setup.py'"'"'; __file__='"'"'/tmp/pip-install-mMxjGq/scipy/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-qiWe3X/install-record.txt --single-version-externally-managed --compile --install-headers /home/pi/oprint/include/site/python2.7/scipy
cwd: /tmp/pip-install-mMxjGq/scipy/
Complete output (152 lines):
Note: if you need reliable uninstall behavior, then install
with pip instead of using `setup.py install`:
- `pip install .`       (from a git repo or downloaded source
release)
- `pip install scipy`   (last SciPy release on PyPI)
Running from scipy source directory.
lapack_opt_info:
lapack_mkl_info:
customize UnixCCompiler
libraries mkl_rt not found in ['/home/pi/oprint/lib', '/usr/local/lib', '/usr/lib', '/usr/lib/arm-linux-gnueabihf']
NOT AVAILABLE
openblas_lapack_info:
customize UnixCCompiler
customize UnixCCompiler
libraries openblas not found in ['/home/pi/oprint/lib', '/usr/local/lib', '/usr/lib', '/usr/lib/arm-linux-gnueabihf']
NOT AVAILABLE
openblas_clapack_info:
customize UnixCCompiler
customize UnixCCompiler
libraries openblas,lapack not found in ['/home/pi/oprint/lib', '/usr/local/lib', '/usr/lib', '/usr/lib/arm-linux-gnueabihf']
NOT AVAILABLE
atlas_3_10_threads_info:
Setting PTATLAS=ATLAS
customize UnixCCompiler
libraries lapack_atlas not found in /home/pi/oprint/lib
customize UnixCCompiler
libraries tatlas,tatlas not found in /home/pi/oprint/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/local/lib
customize UnixCCompiler
libraries tatlas,tatlas not found in /usr/local/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/lib
customize UnixCCompiler
libraries tatlas,tatlas not found in /usr/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/lib/arm-linux-gnueabihf
customize UnixCCompiler
libraries tatlas,tatlas not found in /usr/lib/arm-linux-gnueabihf
<class 'numpy.distutils.system_info.atlas_3_10_threads_info'>
NOT AVAILABLE
atlas_3_10_info:
customize UnixCCompiler
libraries lapack_atlas not found in /home/pi/oprint/lib
customize UnixCCompiler
libraries satlas,satlas not found in /home/pi/oprint/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/local/lib
customize UnixCCompiler
libraries satlas,satlas not found in /usr/local/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/lib
customize UnixCCompiler
libraries satlas,satlas not found in /usr/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/lib/arm-linux-gnueabihf
customize UnixCCompiler
libraries satlas,satlas not found in /usr/lib/arm-linux-gnueabihf
<class 'numpy.distutils.system_info.atlas_3_10_info'>
NOT AVAILABLE
atlas_threads_info:
Setting PTATLAS=ATLAS
customize UnixCCompiler
libraries lapack_atlas not found in /home/pi/oprint/lib
customize UnixCCompiler
libraries ptf77blas,ptcblas,atlas not found in /home/pi/oprint/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/local/lib
customize UnixCCompiler
libraries ptf77blas,ptcblas,atlas not found in /usr/local/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/lib
customize UnixCCompiler
libraries ptf77blas,ptcblas,atlas not found in /usr/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/lib/arm-linux-gnueabihf
customize UnixCCompiler
libraries ptf77blas,ptcblas,atlas not found in /usr/lib/arm-linux-gnueabihf
<class 'numpy.distutils.system_info.atlas_threads_info'>
NOT AVAILABLE
atlas_info:
customize UnixCCompiler
libraries lapack_atlas not found in /home/pi/oprint/lib
customize UnixCCompiler
libraries f77blas,cblas,atlas not found in /home/pi/oprint/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/local/lib
customize UnixCCompiler
libraries f77blas,cblas,atlas not found in /usr/local/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/lib
customize UnixCCompiler
libraries f77blas,cblas,atlas not found in /usr/lib
customize UnixCCompiler
libraries lapack_atlas not found in /usr/lib/arm-linux-gnueabihf
customize UnixCCompiler
libraries f77blas,cblas,atlas not found in /usr/lib/arm-linux-gnueabihf
<class 'numpy.distutils.system_info.atlas_info'>
NOT AVAILABLE
accelerate_info:
NOT AVAILABLE
/home/pi/oprint/local/lib/python2.7/site-packages/numpy/distutils/system_info.py:639: UserWarning:
Atlas (http://math-atlas.sourceforge.net/) libraries not found.
Directories to search for the libraries can be specified in the
numpy/distutils/site.cfg file (section [atlas]) or by setting
the ATLAS environment variable.
self.calc_info()
lapack_info:
customize UnixCCompiler
libraries lapack not found in ['/home/pi/oprint/lib', '/usr/local/lib', '/usr/lib', '/usr/lib/arm-linux-gnueabihf']
NOT AVAILABLE
/home/pi/oprint/local/lib/python2.7/site-packages/numpy/distutils/system_info.py:639: UserWarning:
Lapack (http://www.netlib.org/lapack/) libraries not found.
Directories to search for the libraries can be specified in the
numpy/distutils/site.cfg file (section [lapack]) or by setting
the LAPACK environment variable.
self.calc_info()
lapack_src_info:
NOT AVAILABLE
/home/pi/oprint/local/lib/python2.7/site-packages/numpy/distutils/system_info.py:639: UserWarning:
Lapack (http://www.netlib.org/lapack/) sources not found.
Directories to search for the sources can be specified in the
numpy/distutils/site.cfg file (section [lapack_src]) or by setting
the LAPACK_SRC environment variable.
self.calc_info()
NOT AVAILABLE
Traceback (most recent call last):
File "<string>", line 1, in <module>
File "/tmp/pip-install-mMxjGq/scipy/setup.py", line 492, in <module>
setup_package()
File "/tmp/pip-install-mMxjGq/scipy/setup.py", line 488, in setup_package
setup(**metadata)
File "/home/pi/oprint/local/lib/python2.7/site-packages/numpy/distutils/core.py", line 137, in setup
config = configuration()
File "/tmp/pip-install-mMxjGq/scipy/setup.py", line 395, in configuration
raise NotFoundError(msg)
numpy.distutils.system_info.NotFoundError: No lapack/blas resources found.
----------------------------------------
ERROR: Command errored out with exit status 1: /home/pi/oprint/bin/python2 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-mMxjGq/scipy/setup.py'"'"'; __file__='"'"'/tmp/pip-install-mMxjGq/scipy/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-qiWe3X/install-record.txt --single-version-externally-managed --compile --install-headers /home/pi/oprint/include/site/python2.7/scipy Check the logs for full command output.
Running setup.py install for scipy: finished with status 'error'
The update did not finish successfully. Please consult <code>octoprint.log</code> and <code>plugin_softwareupdate_console.log</code> for details.

**Comment by scottrini** [comment]
There's a new version that might help this, but otherwise check out this comment that should get you sorted: https://github.com/scottrini/OctoPrint-PrusaLevelingGuide/issues/31#issuecomment-779587585

**Comment by ajdavids** [comment]
I'm in version 1.0.12, octopi 0.17, octoprint's latest version, and the oprint venv is already python 3, yet this plug-in fails to update to 1.0.14 because scypi won't install.

**Comment by draeath** [comment]
@ajdavids in your original message I see python2 is being run. Are you *positive* you are using python3 for octoprint?

If you get over that hurdle you may run into #34 as well, if you're unlucky (I'm attempting to work on that one myself)

EDIT: mistaken identity, you are not the fellow who opened the issue and has python2 in the output. Sorry! If you are getting a bunch of errors from gfortran check out #34 - else can you share the output you get when scipy fails to install? It may or may not be the same problem as the original reporter.

**Comment by draeath** [comment]
Tested successfully with:
    numpy==1.20.1
    scipy==1.6.0

Given your code doesn't seem very version dependent, I'm going to put the following in my PR:

    numpy~=1.20.0
    scipy~=1.6.0
 
 This spec should allow anything within 1.20 and 1.6 respectively (ie 1.6.5 would pass but not 1.5.4 or 1.7.3)

**Comment by scottrini** [comment]
Sorry for these issues.  Some changes submitted that I merged added two new dependencies and many users have been having trouble installing them.  So I've rolled back the changes.  If you upgrade to 1.0.15, it's basically identical to 1.0.12 (before the changes), so just upgrade to 1.0.15 and you'll be fine.

I won't merge those changes back in unless we find a way to refactor the code and remove those dependencies.
